addon/services/infinity.js

```diff
@@ -1,9 +1,29 @@
 import Service from '@ember/service';
 import InfinityModel from 'ember-infinity/lib/infinity-model';
+import InfinityPromiseArray from 'ember-infinity/lib/infinity-promise-array';
+import BoundParamsMixin from 'ember-infinity/mixins/bound-params';
 import EmberError from '@ember/error';
-import { isEmpty } from '@ember/utils';
+import { A } from '@ember/array';
+import { isEmpty, typeOf } from '@ember/utils';
 import { scheduleOnce } from '@ember/runloop';
 import { get, set } from '@ember/object';
+import { objectAssign, paramsCheck } from '../utils';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
+
+let findElem = (context) => {
+  let elem;
+  if (
+    context.nodeType === Node.ELEMENT_NODE ||
+    context.nodeType === Node.DOCUMENT_NODE
+  ) {
+    elem = context
+  } else {
+    elem = document.querySelector(context);
+  }
+
+  return elem;
+};
```

Comment:
I'm not sure I understand why the service needs to `findElem` ?

Reply:
Put some documentation.  The request might be from the route or the infinityLoader - might be type Node or String